### PR TITLE
Apollo Server 2.0: AWS Lambda Integration

### PR DESCRIPTION
**Pull request by adnsio**
AWS Lambda integration for Apollo Server 2

Demo: https://4z36935dne.execute-api.eu-west-1.amazonaws.com/production

### CODE REVIEW

File: packages/apollo-server-lambda/src/ApolloServer.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  //provides typings for the integration specific behavior, ideally this would
+  //be propagated with a generic to the super class
+  async createGraphQLServerOptions(
+    event: lambda.APIGatewayProxyEvent,
+    context: lambda.Context,
+  ): Promise<GraphQLOptions> {
+    return super.graphQLServerOptions({ event, context });
+  }
+
+  public async handle({ event, context, callback, gui }: ServerRegistration) {
```
[the remark this answers is not in the document]
Comment (adnsio) [reply packages/apollo-server-lambda/src/ApolloServer.ts]: About CORS for now I copied the old readme example, but we could also add a `cors` option to simplify the setup.

**Comment by adnsio** [comment]
@evans I added the examples in the readme similar to the previous ones.

**Comment by evans** [comment]
@adnsio Thank you so much for the documentation! I think your [comment](https://github.com/apollographql/apollo-server/pull/1188#discussion_r196798819) is spot on. A cors option passed to the `createHandler` that follows the express version https://github.com/expressjs/cors#configuration-options would be great! The `origin` option can probably be a simple string or boolean.

**Comment by adnsio** [comment]
@evans I added the CORS option (like [express](https://github.com/expressjs/cors#configuration-options)) and tests for `lambdaApollo.ts`

**Comment by evans** [comment]
Super awesome work @adnsio! Thank you @bwlt and @gragio for adding code reviews. I'm going to merge this now and include it in the next rc. Adding a deployment guide on Lambda, next to the [Heroku page](https://www.apollographql.com/docs/apollo-server/v2/deployment/heroku.html) would be incredibly helpful and definitely make the package more widespread! It could just be a symlinked copy of the README

**Comment by unicodeveloper** [comment]
@evans I'd handle that 🍾 @adnsio Super thanks for this great work!

**Pull request merged**